File: solutions/NextRound.py
#===============================================
#Method-1: Simple Loop
#===============================================

n, k = map(int, input().split())
a = list(map(int, input().split()))

req_score = a[k-1]
cnt = 0
for x in a:
    if x >= req_score and x > 0: cnt += 1
    else: break
print(count)

#=================================================
#Method-2: Using sum() with generator expression
#=================================================

n, k = map(int, input().split())
a = list(map(int, input().split()))

req_score = a[k-1]
print(sum(1 for x in a if x >= req_score and x > 0))
"""
+In fact we can just write 1 here, we can just write,
+sum(for x in a if x >= req_score and x > 0; here Python takes True = 1 & False = 0
+However mark that, this leads to transversal of the whole array
"""

#==================================================
#Method-3: Count only if 0
#==================================================

#if req_score != 0, k players pass by default
def solve(n, k, a):
    if a[k-1] != 0:
        return(k)
    else:
        return(a.count(0))

n, k = map(int, input().split())
a = list(map(int, input().split()))

ans = solve(n, k, a)
print(ans)

    


        

File: solutions/NextRound.java
import java.util.Scanner;

public class Main {
    static int solve(int n, int k, int[] a) {
        int ans = 0;
        for(int i = 0; i < n; i++) {
            if (a[i] > 0 && a[i] >= a[k-1]) {
                ans += 1;
            }
        }
        return ans;
    }
    
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        
        int n = sc.nextInt();
        int k = sc.nextInt();
        int[] a = new int[n];
        for (int i = 0; i < n; i++) {
            a[i] = sc.nextInt();
        }
        
        int ans = solve(n, k, a);
        System.out.println(ans);
    }
}